Sources/SwiftDocCUtilities/Action/Actions/Convert/ConvertAction.swift

```diff
@@ -452,10 +457,15 @@ public struct ConvertAction: Action, RecreatingContext {
         }
 
         var allProblems = analysisProblems + conversionProblems
-
-        if allProblems.containsErrors == false {
-            let coverageResults = try coverageAction.perform(logHandle: logHandle)
-            allProblems.append(contentsOf: coverageResults.problems)
+        if try context.renderRootModules.isEmpty {
+            convertEncounteredProblems.append(
+                Problem(diagnostic: Diagnostic(
+                    severity: .warning,
+                    identifier: "org.swift.docc.EmptyDoccArchive",
+                    summary: "The generated archive does not contain any pages.",
+                    explanation: "For article-only documentation add a Technology Root using the `TechnologyRoot` directive within a `Metadata` directive (https://www.swift.org/documentation/docc/technologyroot)."
```

Comment:
How do you feel about phrasing this diagnostic summary in terms of what's missing so that the "summary" without the "explanation" provide more information about what the developer needs to do to fix the issue? Perhaps something like this:

```suggestion
                    identifier: "org.swift.docc.MissingTechnologyRoot",
                    summary: "No TechnologyRoot to organize article-only documentation.",
                    explanation: "Article-only documentation need a TechnologyRoot page (indicated by a `TechnologyRoot` directive within a `Metadata` directive) to define the root of the documentation hierarchy."
```

Reply:
I like the rephrasing of this, but I believe that keeping the [TechnologyRoot](https://www.swift.org/documentation/docc/technologyroot) documentation url could be useful for users since it shows an example of how to add the directive.

Reply:
Is there a precedent with adding documentation links in our diagnostics? I am not sure that is something we want to do, it would mean we need to make sure the link remains up to date. @d-ronnqvist what do you think?

Reply:
I don't think we have any other diagnostics with links in the text. 

As an alternative to a link we could add a `Solution` that inserts the missing directives in one of the articles. I'm not sure if that would means that we'd need to associate this diagnostic with one of the files as well. I don't know what would be a good heuristic to pick which source file to associate the diagnostic with. It's possible that something as crude as sorting the file paths alphabetically will suffice for picking an arbitrary but stable file for the solution.

Reply:
Adding the directive automatically would be a great solution for catalogs with only one article, but adding it randomly in a multi-article documentation might result in even more confusion for the user. I will remove the link for now, and we can explore the possibility of adding the directive automatically in another pr 😄